### ISSUE TITLE
Form template inside template created directly via Blaze.renderWithData errors

### ISSUE THREAD

**Issue by freer4** [aldeed/meteor-autoform#789]
I have an application where I'm using `Blaze.renderWithData` to insert some templates with specific data sets. I had originally had this set up to be reactive, with the template data living in one large array I'd append to and save in a session variable. This was problematic because it would reload the entirety of array each time I saved the new to the session variable, which quickly become expensive. 

So, inserting them individually via `Blaze.renderWithData` works and that's all good... except for sometimes with autoform. It appears that autoform is trying to get this element before it exists in the DOM via `AutoForm.templateInstanceForForm` function. I put a log in there to confirm, the function receives the correct ID for the template, but the `getElementById` call returns null. 

I imagine this is Blaze and autoform tripping over each other. I will admit I'm still fairly new to Meteor so I'm not sure of all the inner workings. Is there a way around this that already exists or should this perhaps be a feature update for autoform? 

    Uncaught TypeError: Cannot read property 'formValues' of undefined :: autoform-inputs.js:162 
    markChanged :: autoform-inputs.js:169 
    updateTrackedFieldValue :: afFieldInput.js:72 

I noticed this issue was attached to the innercontext helper: https://github.com/aldeed/meteor-autoform/issues/461. Could be related. 

Upping the setTimeout in afFieldInput.js line 71 to 500ms fixes that error, so likely confirms my theory, but I haven't tested if the delay causes other issues and setTimeout in this context seems hacky to me.

**Comment by javaknight** [comment]
I have updated to the latest version of autoform, and I am now experiencing this same issue.

**Comment by aldeed** [comment]
@freer4 @javaknight, since I don't know how to reliably reproduce this issue, can one of you try to switch the `setTimeout` to `Tracker.afterFlush` and see if that resolves it? Also should probably not pass `formId` to `templateInstanceForForm`.

**Comment by javaknight** [comment]
I will try to make a gist for you this week. I use renderWithData to show an autoform using bootbox.

**Comment by javaknight** [comment]
Well, I cant reproduce this with a simple gist that I created. If you like, I can upload the gist still, but it doesn't demonstrate the error...

**Comment by javaknight** [comment]
Updated it with the code you suggested and I get the following stack trace

Exception from Tracker afterFlush function: Error: There is no current view
    at Object.Blaze._getCurrentView (https://mysite.com/packages/blaze.js?4e49999979a58da0e2265f7bd3f5910f9901b07b:2379:11)
    at Object.Blaze.getView (https://mysite.com/packages/blaze.js?4e49999979a58da0e2265f7bd3f5910f9901b07b:2360:18)
    at Object.AutoForm.viewForForm (https://mysite.com/packages/aldeed_autoform.js?c5125fd9f61c12fcaaca8623c3efa25e438100a2:2751:20)
    at Object.AutoForm.templateInstanceForForm (https://mysite.com/packages/aldeed_autoform.js?c5125fd9f61c12fcaaca8623c3efa25e438100a2:2728:19)
    at https://mysite.com/packages/aldeed_autoform.js?c5125fd9f61c12fcaaca8623c3efa25e438100a2:6976:31
    at Tracker.flush (https://mysite.com/packages/tracker.js?21f0f4306879f57e10ad3a97efe9ea521c5b5775:460:11)

**Comment by freer4** [comment]
This seems inconsistent for me as well, but that sort of makes sense wrapping it in a setTimeout. I'll try the afterFlush now that I'm seeing it again.

**Comment by freer4** [comment]
Call me a newb... I've created a fork and made the change as suggested... but I don't know how to "install" the fork in place of autoform to test it.

**Comment by javaknight** [comment]
freer4: 

1) create a directory called **packages** in your project
2) then inside it, create **meteor-autoform** directory
3) inside that directory put a modified **afFieldInput.js**

meteor should recompile the module with your modified file. if that doesnt work, you may need to unzip the whole code into meteor-autoform directory, and modify the one specific file.

**Comment by freer4** [comment]
Thanks! I am seeing the exact same error you reported upon updating this code.

This (so far) does not seem to have any effect on my app... just throws the error and goes on about its business.

**Comment by aldeed** [comment]
OK, it's not an error we care about in this case, so I think ignoring it makes the most sense:

```js
Tracker.afterFlush(function () {
      var template;
      try {
        template = AutoForm.templateInstanceForForm();
      } catch (e) {}
      if (template) {
        updateTrackedFieldValue(template, c.atts.name);
      }
});
```

**Comment by freer4** [comment]
Is there some reason we would need that template variable? If so, could we attach an on-render to the parent-most template, so that it is set after it is appended to the DOM. Or some other event, I'm not certain on-render isn't being called before DOM insertion.

**Comment by aldeed** [comment]
Should be fixed in 5.1.0
